tests/test_preprocessing/test_columncapper.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+import numpy as np
+import pandas as pd
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.utils import estimator_checks
+from sklearn.utils.estimator_checks import check_transformers_unfitted
+from sklearn.utils.validation import FLOAT_DTYPES
+
+from sklego.common import flatten
+from sklego.preprocessing import ColumnCapper
+from tests.conftest import nonmeta_checks
+
+# ColumnCapper works with nan/inf cells
+nonmeta_checks_allow_nan_inf = [test for test in nonmeta_checks if test.__name__ != 'check_estimators_nan_inf']
+
+
+@pytest.mark.parametrize("test_fn", flatten([
+    nonmeta_checks_allow_nan_inf,
+    # Transformer checks
+    check_transformers_unfitted,
+    # General checks
+    estimator_checks.check_fit2d_predict1d,
+    estimator_checks.check_fit2d_1sample,
+    estimator_checks.check_fit2d_1feature,
+    # ColumnCapper works with 1d arrays. Skipping check_fit1d.
+    # estimator_checks.check_fit1d,
+    estimator_checks.check_get_params_invariance,
+    estimator_checks.check_set_params,
+    estimator_checks.check_dict_unchanged,
+    estimator_checks.check_dont_overwrite_parameters,
+    estimator_checks.check_transformer_general,
+    estimator_checks.check_methods_subset_invariance,
+    estimator_checks.check_transformer_data_not_an_array,
+    estimator_checks.check_transformer_general,
+]))
+def test_estimator_checks(test_fn):
+    test_fn(ColumnCapper.__name__, ColumnCapper())
+
+
+def test_quantiles():
+    def expect_type_error(min_quantile=0.05, max_quantile=0.95):
```

Comment:
never seen this pattern before but i like it.

Reply:
I was improvising. It was my first time implementing automated tests :B

Reply:
good improv, I like it!